src/test/operators/table_scan_sorted_segment_search_test.cpp

```diff
@@ -0,0 +1,106 @@
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+#include "operators/table_scan/sorted_segment_search.hpp"
+#include "storage/segment_iterate.hpp"
+
+namespace opossum {
+
+using TestData = std::tuple<std::string, PredicateCondition, int32_t, std::vector<int32_t>>;
+using Params = std::tuple<TestData, OrderByMode, bool>;
+
+namespace {
+
+auto formatter = [](const ::testing::TestParamInfo<Params> info) {
+  return order_by_mode_to_string.at(std::get<1>(info.param)) + std::get<0>(std::get<0>(info.param)) +
+         (std::get<2>(info.param) ? "WithNulls" : "WithoutNulls");
+};
+
+}  // namespace
+
+class OperatorsTableScanSortedSegmentSearchTest : public BaseTest, public ::testing::WithParamInterface<Params> {
+ protected:
+  void SetUp() override {
+    bool nullable;
+    TestData test_data;
+    std::tie(test_data, _order_by, nullable) = GetParam();
+    std::tie(std::ignore, _predicate_condition, _search_value, _expected) = test_data;
+
+    const bool ascending = _order_by == OrderByMode::Ascending || _order_by == OrderByMode::AscendingNullsLast;
+    const bool nulls_first = _order_by == OrderByMode::Ascending || _order_by == OrderByMode::Descending;
+
+    if (!ascending) {
+      std::reverse(_expected.begin(), _expected.end());
+    }
+
+    _segment = std::make_unique<ValueSegment<int32_t>>(nullable);
+
+    if (nullable && nulls_first) {
+      _segment->append(NULL_VALUE);
+      _segment->append(NULL_VALUE);
+      _segment->append(NULL_VALUE);
+    }
+
+    const auto table_size = 10;
+    for (int32_t row = 0; row < table_size; ++row) {
+      _segment->append(ascending ? row : table_size - row - 1);
+    }
+
+    if (nullable && !nulls_first) {
+      _segment->append(NULL_VALUE);
+      _segment->append(NULL_VALUE);
+      _segment->append(NULL_VALUE);
+    }
+  }
+
+  std::unique_ptr<ValueSegment<int32_t>> _segment;
+  PredicateCondition _predicate_condition;
+  int32_t _search_value;
+  std::vector<int32_t> _expected;
+  OrderByMode _order_by;
+};
+
+INSTANTIATE_TEST_CASE_P(
+    Predicates, OperatorsTableScanSortedSegmentSearchTest,
+    ::testing::Combine(
+        ::testing::Values(
+            // The following rows specify the different testcases.
+            // All of them run on a table that consists of the values from 0 to 9.
+            // The parameters work like this:
+            // 1. predicate condition to use
+            // 2. value to compare
+            // 3. expected result
+            //
+            // Each row is tested with all four sorted orders and nullable or non-nullable segments. For descending
+            // sort orders, the segments contain the values from 9 to 0 and the expected result is reversed, so that
+            // you only need to specify the expected result in ascending order.
+            TestData("Equals", PredicateCondition::Equals, 5, {5}),
```

Comment:
one thing is missing here imo, and that is that a EQUALS scan matches multiple rows.